distributed/client.py

```diff
@@ -650,13 +650,19 @@ def __init__(
 
         if isinstance(address, (rpc, PooledRPCCall)):
             self.scheduler = address
-        elif hasattr(address, "scheduler_address"):
+        elif isinstance(getattr(address, "scheduler_address", None), str):
             # It's a LocalCluster or LocalCluster-compatible object
             self.cluster = address
             with suppress(AttributeError):
                 loop = address.loop
             if security is None:
                 security = getattr(self.cluster, "security", None)
+        elif address is not None and not isinstance(address, str):
+            raise ValueError(
+                "Scheduler address must be a string or a Cluster instance, got {}".format(
+                    address
```

Comment:
Maybe we just want to report back to the user the type of object we were passed, not the actual object.

```suggestion
                    type(address)
```

Reply:
good thought - pushed the recommended change and the CI build now passes.